### ISSUE TITLE
Datalist: update aria and a11y considerations

### ISSUE THREAD

**Issue by jessiehuff** [patternfly/patternfly-react#4896]
**Describe the issue. What is the expected and unexpected behavior?**
We should take a deeper look at the datalist component and determine what the right aria for each scenario would be. Roles were added to communicate the selectable datalist, but they are currently missing `aria-activedescendant` when an option is visually focused and `aria-selected/aria-multiselectable` when an item is selected as this [WAI example](https://www.w3.org/TR/2017/WD-wai-aria-practices-1.1-20170628/examples/listbox/listbox.html) shows. However, I question whether this should be the case for all datalist examples.

**Please provide the steps to reproduce. Feel free to link CodeSandbox or another tool.**

Open up VO (cmd + F5), go to datalist example and see that it considers all datalist examples as selectable now which isn't accurate. It also doesn't communicate to users when they have focus on an option or when they've selected an item on the selectable datalist (due to the missing aria).

**Is this a bug or enhancement? If this issue is a bug, is this issue blocking you or is there a work-around?**
a11y bug
